src/switchroot/ostree-mount-util.h

```diff
@@ -106,6 +128,115 @@ read_proc_cmdline_ostree (void)
   return ret;
 }
 
+/* If the key matches the start of the line, strip the key from the line just
+ * to get the value
+ */
+static inline bool
+rm_key_if_match (char *line, const char *key, const size_t key_len, char *out)
+{
+  if (strstr (line, key) == line)
+    {
+      char *value = line + key_len;
+      strcpy (out, value);
+      return true;
+    }
+
+  return false;
+}
+
+/* On completion version and options will be set to new values if the version
+ * is more recent. Will loop line through line on the passed in open FILE.
+ */
+static inline int
+copy_if_higher_version (FILE *f, char *version, char *options)
+{
+  char __attribute__ ((cleanup (free_char))) *line = NULL;
+  char version_local[PATH_MAX] = "";
+  char options_local[PATH_MAX] = "";
+  /* Note getline() will reuse the previous buffer */
+  for (size_t len = 0; getline (&line, &len, f) != -1;)
+    {
+      /* This is an awful hack to avoid depending on GLib in the
+       * initramfs right now.
+       */
+      if (rm_key_if_match (line, "version ", sizeof ("version"), version_local))
```

Comment:
Please pass in sizeof(version_local) to ensure we're not overflowing the destination.

Reply:
I guess you could theoretically overflow version_local if the version field in "line" was longer that 4096 bytes. That would be nuts, but no hard checking, I'll do similar for options_local, since it's the exact same logic.

Reply:
I actually did this a different way now, I didn't need to copy strings at all, I could just the pointer to the string allocated by getline and if you set the line char* to NULL, getline will allocate a new buffer next time around.

I copy the pointer to a string instead of strcpy now.